File: challenge-03/main.py
import re
failedPasswords = list()
with open('challenge-03/data.txt','r') as f:
    for row in f.readlines():
        info = row.split(' ')
        range = info[0].split('-')
        searchedVal = info[1][:-1]
        password = info[2]

        regexSearch = searchedVal +'{'+range[0]+','+range[1]+'}'
        if re.search(regexSearch, password):
            continue
        failedPasswords.append(password)
print(failedPasswords[41])

File: challenge-03/main.dart
import 'dart:io';

void main() {
  File('challenge-03/data.txt').readAsLines().then((List<String> rows) {
    final failedPasswords = [];
    for (final row in rows) {
      final info = row.split(' ');
      final range = info[0].split('-');
      final searchVal = info[1].substring(0, info[1].length - 1); // remove :
      final password = info[2];

      final regexString = '$searchVal{${range[0]},${range[1]}}';
      final regex = RegExp(r'' + regexString + '');
      if (!regex.hasMatch(password)) {
        failedPasswords.add(password);
      }
    }
    print(failedPasswords[41] ?? '');
  }).onError((_, __) {
    print('archivo inexistente');
  });
}
